sphinxext/icon.py

```diff
@@ -1,121 +1,337 @@
+    ICONS = set([
+        'caret-right',
+        'caret-square-left',
+        'caret-square-right',
+        'check-circle',
+        'check-square',
+        'copy',
+        'edit',
+        'exclamation-circle',
+        'eye',
+        'globe',
+        'lock',
+        'pencil-alt',
+        'save',
+        'star',
+        'sync-alt',
+        'trash',
+        'trash-alt',
+        'users'
+    ])
+
+    # Discouraged, but not prohibited (yet)
+    ICONS.add('arrow-right')
+    ICONS.add('asterisk')
+    ICONS.add('check')
+    ICONS.add('ellipsis-h')
+    ICONS.add('minus')
+    ICONS.add('plus')
+
+
+class IconsBrandsFA5:
+    """FontAwesome 5 Brand icon font"""
+    NODE_NAME = 'IconBrandFA5'
+    ROLE_NAMES = ('icon-fa5-brands', 'iconb')
+    CSS_PREFIX = 'fab fa-'
+    ICONS = set([
+        'windows',
+    ])
+
+
+class IconsFA4:
+    """FontAwesome 4.7 icon font"""
+    NODE_NAME = 'IconFA4'
+    ROLE_NAMES = ('icon-fa4',)
+    CSS_PREFIX = 'fa4 fa4-'
+    ICONS = set([
+        'book',
+        'caret-down',
+        'caret-right',
+        'circle',
+        'download',
+        'enabled-banner-container',
+        'exclamation-circle',
+        'exclamation-triangle',
+        'globe',
+        'info',
+        'info-circle',
+        'lock',
+        'pencil',
+        'plus',
+        'plus-square',
+        'question-circle',
+        'refresh',
+        'search',
+        'stop',
+        'times-circle',
+        'trash-o'
+    ])
+
+
+class IconsMMS:
+    """MMS icon font"""
+    NODE_NAME = 'IconMMS'
+    ROLE_NAMES = ('icon-mms',)
+    CSS_PREFIX = 'mms-icon mms-icon-'
+    ICONS = set([
+        '2fa',
+        'activity',
+        'add',
+        'addcenter',
+        'api',
+        'auth',
+        'back',
+        'bell',
+        'check',
+        'cloud',
+        'configsvr-startup2',
+        'continuous',
+        'creditcard',
+        'dashboard',
+        'database',
+        'databases',
+        'deadface',
+        'dragleft',
+        'dragtopleft',
+        'edit',
+        'ellipsis',
+        'floppy',
+        'graph',
+        'grid',
+        'hammer',
+        'laptop',
+        'list',
+        'list-skinny',
+        'lock',
+        'logo-amazon',
+        'logo-apple',
+        'logo-linux',
+        'logo-redhat',
+        'logo-ubuntu',
+        'logo-windows',
+        'medium-cloud',
+        'metrics',
+        'modify',
+        'office',
+        'ops-manager',
+        'paused',
+        'pointintime',
+        'remove',
+        'replica-set-configsvr',
+        'rocketbot',
+        'sadface',
+        'servers',
+        'settings',
+        'setup',
+        'smartphone',
+        'ssl',
+        'startup2',
+        'support1',
+        'support2',
+        'surprisedface',
+        'topology',
+        'umbrella'
+    ])
+
+    for mode in ('',) + ALL_SUFFIXES:
+        ICONS.add('wrench' + mode)
+
+    for mode in ('', '-add', '-edit', '-remove', '-restart'):
```

Comment:
Again this part is pretty clear but seems not so DRY (perhaps that's not a problem).

We could use `ICONS.update` here as well:
```python
for mode in ('', '-add', '-edit', '-remove', '-restart'):
    ICONS.update([icon + mode for icon in [
        'standalone',
        'replica-set',
        'cluster',
        'mongos',
    ]])
```
or this more slightly more functional approach
```python
for mode in ('', '-add', '-edit', '-remove', '-restart'):
    def mode_icon(icon):
        return icon + mode
    def add_mode_icons(icons):
        ICONS.update([mode_icon(icon) for icon in icons])
    
    # Agents
    add_mode_icons([
        'monitoring',
        'backup',
        'automation'
    ]

    # Deployment Items
    add_mode_icons([
        'standalone',
        'replica-set',
        'cluster',
        'mongos'
    ])
```

Reply:
Both of these are invalid in a class context in Python, unfortunately. Refactored to allow this